actsims/batchTools.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np, os
+from pixell import enmap, fft, powspec, curvedsky
+from . import flipperDict, simTools
+import healpy as hp
+
+
+input_param_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../inputParams')
+input_path      = lambda x : os.path.join(input_param_dir, x)
+
+signal_dict     = flipperDict.flipperDict()
+noise_dict      = flipperDict.flipperDict()
+signal_dict.read_from_file(input_path('signal.dict')) 
+noise_dict.read_from_file(input_path('templateInputsMr3c.dict'))
+
+cmb_dir         = os.path.join(os.path.dirname(os.path.abspath(__file__)), signal_dict['cmbDir'])
+data_dir        = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../data')
+
+
+class MR3PATCH_HELPER(object):
+    # a helper class to quickly generate sims for given patch
+    def __init__(self, set_idx, sim_idx, patch, cmb_type='LensedCMB', noisediag_only=False, dobeam=True, add_foregrounds=True, apply_window=True):
+        self.patch    = patch.lower()
+        assert(self.patch in ['boss', 'deep1', 'deep56', 'deep5', 'deep6', 'deep8'])
+        assert(cmb_type in ['LensedCMB', 'UnlensedCMB', 'LensedUnabberatedCMB'])
+
```

Comment:
I set things up so that that dict files, not python code, define what are the allowed patch names.  (For instance there is a dict file to give the settings to make planck sims for Omar).  So I don't think this adheres to that philosophy.

Reply:
I think it sometimes useful to list the allowed user inputs at the top of functions as assertion statements. (1) First, you can capture bad user inputs earlier rather than later (2) Second, it is a way for other developers to see what is the expected inputs without any comments or need to figure it out from the configuration files or file names. 

I will take out the first assertion statement. I think that's pretty clear. But I don't think the cmb_types are listed in any dict file. If we want to put them in the dictionary file, I am fine with that.

Reply:
OK, sounds good, thanks!